zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -438,6 +438,41 @@ public void createPersistent(String path)
     createPersistent(path, false);
   }
 
+  /**
+   * Create a persistent node with TTL.
+   * @param path the path where you want the node to be created
+   * @param ttl TTL of the node in milliseconds
+   * @throws ZkInterruptedException
+   *           if operation was interrupted, or a required reconnection got interrupted
+   * @throws IllegalArgumentException
+   *           if called from anything except the ZooKeeper event thread
+   * @throws ZkException
+   *           if any ZooKeeper exception occurred
+   * @throws RuntimeException
+   *           if any other exception occurs
+   */
+  public void createPersistentWithTTL(String path, long ttl)
```

Comment:
Do we need `public void createPersistentSequentialWithTTL(String path, long ttl)` and `public void createPersistentSequentialWithTTL(String path, boolean createParents, long ttl)`?

Reply:
I created them because the version without TTL existed, and the version with TTL is now supported in Zookeeper. If you think of a specific reason to not have them, please let me know and I can remove them.

Reply:
Sorry for the confusion, I shouldn't have put the comment on this line. But I was actually asking about the sequential nodes, when I compared them I found that we have different methods for persistentWithTTL vs persistentSequentialWithTTL.

Reply:
I see your point now. for the createParents version, I think it doesn't make sense for sequential mode because the path is altered, and wouldn't be clear what the complete path of the ZNode will be at the end. For the other API you asked about, I think the data argument can be set to null to achieve the same behavior. I tried to be consistent with the existing API, so I only added new methods that correspond to existing ones.